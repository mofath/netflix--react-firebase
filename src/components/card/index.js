import React, { useState, useContext, createContext } from 'react';

import { Container, Group, Title, SubTitle, Text, Item, Image, Enteties, Maturity, Content, FeatureClose, FeatureText, Feature, Meta, FeatureTitle } from './styles/card'

export const FeatureContext = createContext();

export default function Card({ children, ...restProps }) {
    const [ShowFeature, setShowFeature] = useState(false);
    const [ItemFeature, setItemFeature] = useState({});

    return (
        <FeatureContext.Provider value={{ ShowFeature, setShowFeature, ItemFeature, setItemFeature }}>
            <Container {...restProps}>{children}</Container>
        </FeatureContext.Provider>
    )
};

Card.Group = function cardGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
};

Card.Title = function cardTitle({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>
};

Card.SubTitle = function cardSubTitle({ children, ...restProps }) {
    return <SubTitle {...restProps}>{children}</SubTitle>
};

Card.Text = function cardText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
};

Card.Meta = function cardMeta({ children, ...restProps }) {
    return <Meta {...restProps}>{children}</Meta>
};


Card.Item = function CardItem({ item, children, ...restProps }) {
    const { setShowFeature, setItemFeature } = useContext(FeatureContext);

    return (
        <Item
            onClick={() => {
                setItemFeature(item);
                setShowFeature(true);
            }}
            {...restProps}
        >
            {children}
        </Item>
    );
};

Card.Enteties = function cardEnteties({ children, ...restProps }) {
    return <Enteties {...restProps} >{children}</Enteties>
}

Card.Image = function cardImage({ src, ...restProps }) {
    return <Image src={src} {...restProps} />
}

Card.Maturity = function cardMaturity({ children, ...restProps }) {
    return <Maturity {...restProps} >{children}</Maturity>
}

Card.Content = function cardContent({ children, ...restProps }) {
    return <Content {...restProps} >{children}</Content>
}

Card.Feature = function CardFeature({ children, category, ...restProps }) {
    const { ShowFeature, setShowFeature, ItemFeature } = useContext(FeatureContext);

    return ShowFeature ? (
        <Feature {...restProps} src={`images/${category}/${ItemFeature.genre}/${ItemFeature.slug}/large.jpg`} >
            <Content>
                <FeatureTitle>{ItemFeature.title}</FeatureTitle>
                <FeatureText>{ItemFeature.description}</FeatureText>
                <FeatureClose onClick={() => setShowFeature(false)}>
                    <img src='/images/icons/close.png' alt="Close" />
                </FeatureClose>
                <Group margin="30px 0" flexDirection="row" alignItems="center">
                    <Maturity rating={ItemFeature.maturity}>
                        {ItemFeature.maturity < 12 ? 'PG' : ItemFeature.maturity}
                    </Maturity>
                    <FeatureText fontWeight="bold">
                        {ItemFeature.genre.charAt(0).toUpperCase() + ItemFeature.genre.slice(1)}
                    </FeatureText>
                </Group>
                {children}
            </Content>
        </Feature >
    ) :
        null
}