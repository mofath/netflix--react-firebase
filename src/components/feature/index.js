import React from 'react';

import { Container, Title, SubTitle } from './styles/feature';


export default function Feature({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}

Feature.Title = function featureTitle({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>
}

Feature.SubTitle = function featureSubTitle({ children, ...restProps }) {
    return <SubTitle {...restProps}>{children}</SubTitle>
}