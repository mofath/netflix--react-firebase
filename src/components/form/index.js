import React from 'react';

import { Container, Title, Text, SubText, Error, Link, Input, Submit, Base } from './styles/from';


export default function Form({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
};

Form.Input = function fromInput({ children, ...restProps }) {
    return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function fromSubmit({ children, ...restProps }) {
    return <Submit {...restProps}>{children}</Submit>;
};

Form.Error = function fromError({ children, ...restProps }) {
    return <Error {...restProps}>{children}</Error>;
};

Form.Base = function fromBase({ children, ...restProps }) {
    return <Base {...restProps}>{children}</Base>;
};


Form.Title = function fromTitle({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>;
};

Form.Text = function fromText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>;
};

Form.SubText = function fromSubText({ children, ...restProps }) {
    return <SubText {...restProps}>{children}</SubText>;
};

Form.Link = function fromLink({ children, ...restProps }) {
    return <Link {...restProps}>{children}</Link>;
};