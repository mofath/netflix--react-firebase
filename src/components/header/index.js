import React, { useState } from 'react';

import { Link as ReachRouterLink } from 'react-router-dom';

import { Background, Container, Logo, ButtonLink, Feature, Text, FeatureCallOut, Link, Group, Picture, Dropdown, Profile, Search, SearchIcon, SearchInput } from './styles/header'

export default function Header({ bg = true, children, ...restProps }) {
    return bg ? (
        <Background data-testid="header-bg" {...restProps}>
            {children}
        </Background>
    ) : (
            <Container>
                {children}
            </Container>
        );
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
    return <Feature>{children}</Feature>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
    return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
    return (
        <ReachRouterLink to={to}>
            <Logo {...restProps} />
        </ReachRouterLink>
    );
};


Header.Text = function HeaderText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
    return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.TextLink = function headerTextLink({ children, ...restProps }) {
    return <Link {...restProps}>{children}</Link>
}

Header.Group = function headerGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
}

Header.Picture = function headerPicture({ src, ...restProps }) {
    return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.Dropdown = function headerDropdown({ children, ...restProps }) {
    return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.Profile = function headerProfile({ children, ...restProps }) {
    return <Profile {...restProps}>{children}</Profile>
}

Header.Search = function HeaderSearch({ searchTerm, setSearchTerm, ...restProps }) {
    const [SearchActive, setSearchActive] = useState(false);

    return (
        <Search {...restProps} >
            <SearchIcon onClick={() => setSearchActive(SearchActive => !SearchActive)} >
                <img src='images/icons/search.png' alt="search" />
            </SearchIcon>
            <SearchInput
                value={searchTerm}
                onChange={({ target }) => setSearchTerm(target.value)}
                placeholder="Search films and series"
                active={SearchActive}
            />
        </Search>
    )
}