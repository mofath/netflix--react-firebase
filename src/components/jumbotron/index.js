import React from 'react';

import { Inner, Item, Container, Pane, Title, Subtitle, Image } from './styles/jumbotron';

export default function Jumbotron({ children, direction = 'row', ...restProps }) {
    return (
        <Item {...restProps}>
            <Inner direction={direction} >
                {children}
            </Inner>
        </Item>
    )
}

Jumbotron.Container = function JumbotronConainer({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}

Jumbotron.Pane = function JumbotronConainer({ children, ...restProps }) {
    return <Pane {...restProps}>{children}</Pane>
}


Jumbotron.Title = function JumbotronConainer({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>
}

Jumbotron.Subtitle = function JumbotronConainer({ children, ...restProps }) {
    return <Subtitle {...restProps}>{children}</Subtitle>
}

Jumbotron.Image = function JumbotronConainer({ ...restProps }) {
    return <Image  {...restProps} />
}

