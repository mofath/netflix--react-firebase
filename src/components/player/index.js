import React, { useState, createContext, useContext } from 'react';

import { Cotainer, Inner, Button, Overlay, Close } from './styles/player';
import ReactDOM from 'react-dom';

export const PlayerContext = createContext();

export default function Player({ children, ...restProps }) {
    const [ShowPlayer, setShowPlayer] = useState(false);
    return (
        <PlayerContext.Provider value={{ ShowPlayer, setShowPlayer }}>
            <Cotainer {...restProps}>{children}</Cotainer>
        </PlayerContext.Provider>
    )
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
    const { ShowPlayer, setShowPlayer } = useContext(PlayerContext)
    return ShowPlayer ? ReactDOM.createPortal(
        <Overlay onClick={() => setShowPlayer(false)} data-testid="player">
            <Inner>
                <video id="netflix-player" controls>
                    <source src={src} type="video/mp4" />
                </video>
                <Close />
            </Inner>
        </Overlay>,
        document.body
    )
        : null;
};

Player.Button = function PlayerButton({ ...restProps }) {
    const { setShowPlayer } = useContext(PlayerContext);

    return (
        <Button onClick={() => setShowPlayer((ShowPlayer) => !ShowPlayer)} {...restProps}>
            Play
        </Button>
    );
};