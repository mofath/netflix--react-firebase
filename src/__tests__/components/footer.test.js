import React from "react";
import { render } from "@testing-library/react";
import { Footer } from "../../components";

describe("<Footrt />", () => {
  it("renders the <Footer /> with populated data", () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Contact us.</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#">FAQ</Footer.Link>
            <Footer.Link href="#">Investors Relations</Footer.Link>
            <Footer.Link href="#">Ways To Watch</Footer.Link>
            <Footer.Link href="#">Coporate Information</Footer.Link>
            <Footer.Link href="#">Netflix Originals</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Help Center</Footer.Link>
            <Footer.Link href="#">Jobs</Footer.Link>
            <Footer.Link href="#">Terms of Use</Footer.Link>
            <Footer.Link href="#">Contact Us</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Account</Footer.Link>
            <Footer.Link href="#">Redeem Gift Cards</Footer.Link>
            <Footer.Link href="#">Privacy</Footer.Link>
            <Footer.Link href="#">Speed Test</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Media Center</Footer.Link>
            <Footer.Link href="#">Redeem Gift Cards</Footer.Link>
            <Footer.Link href="#">Privacy</Footer.Link>
            <Footer.Link href="#">Help Center</Footer.Link>
          </Footer.Column>

          <Footer.Text>Netflix Cairo, Egypt</Footer.Text>
        </Footer.Row>
      </Footer>
    );

    expect(getByText("Questions? Contact us.")).toBeTruthy();
    expect(getByText("FAQ")).toBeTruthy();
    expect(getByText("Investors Relations")).toBeTruthy();
    expect(getByText("Ways To Watch")).toBeTruthy();
    expect(getByText("Coporate Information")).toBeTruthy();
    expect(getByText("Netflix Originals")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
