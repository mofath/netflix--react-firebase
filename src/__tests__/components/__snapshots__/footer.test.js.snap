// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footrt /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-qu6i6-0 kBwVlK"
>
  <p
    class="footer__Title-qu6i6-3 costGr"
  >
    Questions? Contact us.
  </p>
  <div
    class="footer__Break-qu6i6-6 cnYatT"
  />
  <div
    class="footer__Row-qu6i6-2 eQpqDY"
  >
    <div
      class="footer__Column-qu6i6-1 jAYoWp"
    >
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Investors Relations
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Ways To Watch
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Coporate Information
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-1 jAYoWp"
    >
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Help Center
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Contact Us
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-1 jAYoWp"
    >
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Redeem Gift Cards
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-qu6i6-1 jAYoWp"
    >
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Media Center
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Redeem Gift Cards
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-qu6i6-5 dthowc"
        href="#"
      >
        Help Center
      </a>
    </div>
    <p
      class="footer__Text-qu6i6-4 czHFYg"
    >
      Netflix Cairo, Egypt
    </p>
  </div>
</div>
`;
